calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -0,0 +1,309 @@
+        """Cleanup after a pod."""
+        self.pod_name = args[3].replace('-', '_')
+        self.docker_id = args[4]
+
+        # Remove the profile for the workload.
+        calicoctl('container', 'remove', self.docker_id)
+        calicoctl('profile', 'remove', self.pod_name)
+
+    def _configure_interface(self):
+        """Configure the Calico interface for a pod."""
+        container_ip = self._read_docker_ip()
+        self._delete_docker_interface()
+        print('Configuring Calico networking.')
+        ep = container_add(self.docker_id, container_ip, 'eth0')
+        interface_name = generate_cali_interface_name(IF_PREFIX, ep.endpoint_id)
+        node_ip = self._get_node_ip()
+        print('Adding IP %s to interface %s' % (node_ip, interface_name))
+        check_call(['ip', 'addr', 'add', node_ip + '/32',
+                    'dev', interface_name])
+        print('Finished configuring network interface')
+        return ep
+
+    def _get_node_ip(self):
+        """Get the IP for the host node from the k8s API."""
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        s.connect(('8.8.8.8', 80))
+        ip = s.getsockname()[0]
+        s.close()
+        return ip
+
+    def _read_docker_ip(self):
+        """Get the IP for the pod's infra container."""
+        ip = check_output([
+            'docker', 'inspect', '-format', '{{ .NetworkSettings.IPAddress }}',
+            self.docker_id
+        ])
+        # Clean trailing whitespace (expect a '\n' at least).
+        ip = ip.strip()
+
+        print('Docker-assigned IP was %s' % ip)
+        return ip
+
+    def _delete_docker_interface(self):
+        """Delete the existing veth connecting to the docker bridge."""
+        print('Deleting eth0')
+
+        # Get the PID of the container.
+        pid = check_output([
+            'docker', 'inspect', '-format', '{{ .State.Pid }}',
+            self.docker_id
+        ])
+        # Clean trailing whitespace (expect a '\n' at least).
+        pid = pid.strip()
+
+        # Set up a link to the container's netns.
+        print(check_output(['mkdir', '-p', '/var/run/netns']))
+        netns_file = '/var/run/netns/' + pid
+        if not os.path.isfile(netns_file):
+            print(check_output(['ln', '-s', '/proc/' + pid + '/ns/net',
+                                netns_file]))
+
+        # Reach into the netns and delete the docker-allocated interface.
+        print(check_output(['ip', 'netns', 'exec', pid,
+                            'ip', 'link', 'del', 'eth0']))
+
+        # Clean up after ourselves (don't want to leak netns files)
+        print(check_output(['rm', netns_file]))
+
+    def _configure_profile(self, endpoint):
+        """
+        Configure the calico profile for a pod.
+
+        Currently assumes one pod with each name.
+        """
+        calicoctl('profile', 'add', self.pod_name)
+        pod = self._get_pod_config()
+        profile_name = self.pod_name
```

Comment:
Move this line to the top so that we can use profile_name everywhere that expects a profile_name, and we can replace self.pod_name in the calicoctl call above.